website/api/recursion.md

```diff
@@ -0,0 +1,148 @@
+1. Each _Segment_ is proven, resulting in a _SegmentReceipt_.
+1. Each _SegmentReceipt_ is _lifted_, resulting in a _SuccinctReceipt_.
+1. Pairs of _SuccinctReceipts_ are _joined_, resulting in another _SuccinctReceipt_. This continues until a single _SuccinctReceipt_ remains.
+1. The final _SuccinctReceipt_ is passed through _identity_p254_, which prepares for Groth16 proving.
+1. The _SuccinctReceipt_ is _compressed_, generating a _Groth16Receipt_.
+
+The _Groth16Receipt_ can now be posted on-chain and verified by the [RISC Zero Verifier Contract].
+
+## Recursive Circuit Architecture
+
+RISC Zero's zkVM consists of three circuits.
+
+1. The RISC-V Circuit is a STARK circuit that proves correct execution of RISC-V programs.
+1. The Recursion Circuit is a separate STARK circuit, that's designed to efficiently generate proofs for the verification of STARK proofs and to support the integration of custom accelerator circuits into the zkVM.
+   This circuit has a similar architecture to the RISC-V Circuit, but with fewer columns and a much simpler instruction set.
+   The same [proof system] is used for both the RISC-V Circuit and the Recursion Circuit.
+1. The STARK-to-SNARK Circuit is an R1CS circuit that proves verification of proofs from the Recursion Circuit.
+
+## Recursion Programs
+The Recursion Circuit supports a number of programs, including `lift()`, `join()`, `resolve()`, and `identity_p254()`.
+Instead, users should access recursion via the [Prover].
+
+To aid in conceptual understanding, we offer the following explanations of the recursion programs:
+
+1. The `lift()` program verifies a STARK proof from the RISC-V Prover, using the Recursion Prover. This recursion proof has a single constant-time verification procedure, with respect to the original segment length, and is then used as the input to all other recursion programs (e.g. join, resolve, and identity_p254).
+
+1. The `join()` program verifies two proofs from the Recursion Prover, using the Recursion Prover. By repeated application of `join()`, any number of receipts for execution spans within the same session can be compressed into a single receipt for the entire session.
+
+1. The `identity_p254()` program verifies a proof from the Recursion Prover using the Poseidon254 hash function. The identity_p254 program is used as the last step in the prover pipeline before running the Groth16 prover.
+
+## STARK-to-SNARK Wrapping
+All of the recursion programs in the previous section output a [SuccinctReceipt], which is a STARK proof (~200kB)
+
+The final step in the recursion process is `compress()`, which outputs a [Groth16Receipt], which can be verified on-chain using the [RISC Zero Verifier Contract].
+
+[composite, succinct or groth16 receipts]: https://docs.rs/risc0-zkvm/1.0/risc0_zkvm/enum.ReceiptKind.html
+[Groth16Receipt]: https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.Groth16Receipt.html
+[proof system]: ../docs/proof-system/proof-system-sequence-diagram.md
```

Comment:
This will break when we promote the version from next to 1.0.

If you are making links from 'unversioned' to 'versioned' or vice-versa (e.g. /api -> /docs) then you need to use absolute links rather than relative ones.

This section tries to describe the reasoning: https://docusaurus.io/docs/versioning#recommended-practices